email/ncmec.go

```diff
@@ -0,0 +1,273 @@
+package email
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"gitlab.com/NebulousLabs/errors"
+)
+
+const (
+	// ncmecBaseURI is the base URI for NCMEC's production API.
+	ncmecBaseURI = "https://report.cybertip.org/ispws"
+
+	// ncmecTestBaseURI is the base URI for NCMEC's test API.
+	ncmecTestBaseURI = "https://exttest.cybertip.org/ispws"
+)
+
+type (
+
+	// NCMECCredentials holds the credentials that are required to authenticate
+	// with NCMEC's API.
+	NCMECCredentials struct {
+		Username string
+		Password string
+		Debug    bool
```

Comment:
This one could use a comment explaining what it does.

Reply:
updated